templates/microApp/nginx.conf

```diff
@@ -0,0 +1,92 @@
+user  nginx;
+worker_processes  auto;
+
+error_log  /var/log/nginx/error.log  notice;
+pid        /var/run/nginx.pid;
+
+events {
+    worker_connections  1024;  ## Default: 1024
+}
+
+http {
+    include       /etc/nginx/mime.types;
+    default_type  application/octet-stream;
+
+    log_format  main  '$remote_addr - $remote_user [$time_local] "$request" '
+                      '$status $body_bytes_sent "$http_referer" '
+                      '"$http_user_agent" "$http_x_forwarded_for"';
+
+    access_log  /var/log/nginx/access.log  main;
+    sendfile        on;
+    #tcp_nopush     on;
+
+    #keepalive_timeout  0; #The zero value disables keep-alive client connections.
+    keepalive_timeout  65;
+
+    # Compression
+    # Enable Gzip compressed.
+    gzip on;
+
+    # Compression level (1-9).
+    # 5 is a perfect compromise between size and cpu usage, offering about
+    # 75% reduction for most ascii files (almost identical to level 9).
+    gzip_comp_level    5;
+
+    # Don't compress anything that's already small and unlikely to shrink much
+    # if at all (the default is 20 bytes, which is bad as that usually leads to
+    # larger files after gzipping).
+    gzip_min_length    256;
+
+    # Compress data even for clients that are connecting to us via proxies,
+    # identified by the "Via" header (required for CloudFront).
+    gzip_proxied       any;
+
+    # Tell proxies to cache both the gzipped and regular version of a resource
+    # whenever the client's Accept-Encoding capabilities header varies;
+    # Avoids the issue where a non-gzip capable client (which is extremely rare
+    # today) would display gibberish if their proxy gave them the gzipped version.
+    gzip_vary          on;
+
+    # This should be turned on if you are going to have pre-compressed copies (.gz) of 
+    # static files available. If not it should be left off as it will cause extra I/O 
+    # for the check. It is best if you enable this in a location{} block for 
+    # a specific directory, or on an individual server{} level. 
+    # gzip_static on; 
+
+    map $sent_http_content_type $expires {
+        default                    off;
+        text/html                  epoch;
+        application/json           1d;
+        text/css                   365d;
+        application/javascript     365d;
+        ~image/                    365d;
+        ~font/                     365d;
+    }
+
+    server {
+        listen  80;
+        listen  [::]:80;  
+        # change the domain name
+        server_name  domain.name;
+        expires $expires;
```

Comment:
Don't use expire header, use cache control. 

There will be 2 diff rules. 
1. For static assets: 1 year TTL. 
2. For pages and menifest.json: 15 minutes.

Reply:
The expires mentioned in line no 71 is not an header but an nginx directive. This directive adds cache-control header according to the rules as mentioned in map in line no. 56.

References: 
1. [Digital Ocean Blog](https://www.digitalocean.com/community/tutorials/how-to-implement-browser-caching-with-nginx-s-header-module-on-ubuntu-20-04#step-3-%E2%80%94-configuring-cache-control-and-expires-headers)
2. [Nginx Docs](http://nginx.org/en/docs/http/ngx_http_headers_module.html)

Reply:
Following results were when using the above directive:

- `Cache-control: no-cache` header for html files
![image](https://user-images.githubusercontent.com/24247542/135607699-e7396281-c628-41e0-89ec-0c00984aa20c.png)

- `Cache-control : max-age=31536000` header for jpg files
![image](https://user-images.githubusercontent.com/24247542/135607733-0a1a4135-d68c-405f-9daf-5fd94cf2d70d.png)

Reply:
Cache control on static assets should be 

cache-control: public, max-age=31535946

We should not be using Expires at all. Only one of them makes sense. 

ETag is already added and should be kept as is.

Reply:
Updated the pull request. 
- Removed the `expires` header.
- Added `public` to the `Cache-Control` header
- html and json files now have 15min max-age
- Static files have 1hr max-age

![image](https://user-images.githubusercontent.com/24247542/135862704-e70f86c4-3cf1-420f-af46-bcde74f228ed.png)